### PR TITLE
Re-add xp orb merging patches

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: patches/server/0427-ExperienceOrbMergeEvent.patch [hunk 1]
```diff
@@ -2,21 +2,22 @@ From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Aikar <aikar@aikar.co>
 Date: Tue, 19 Dec 2017 22:57:26 -0500
 Subject: [PATCH] ExperienceOrbMergeEvent
+
 Has to be reimplemented at one point maybe
```
Comment (Prof-Bloodstone) [inline patches/server/0427-ExperienceOrbMergeEvent.patch:6]: I believe this should be removed

File: patches/server/0427-ExperienceOrbMergeEvent.patch [hunk 2]
```diff
@@ -2,21 +2,22 @@ From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
...
+@@ -620,7 +620,7 @@ public class CraftEventFactory {
                      if (e instanceof net.minecraft.world.entity.ExperienceOrb) {
                          net.minecraft.world.entity.ExperienceOrb loopItem = (net.minecraft.world.entity.ExperienceOrb) e;
                          // Paper start
--                        if (!loopItem.removed && !(maxValue > 0 && loopItem.value >= maxValue)) {
-+                        if (!loopItem.removed && !(maxValue > 0 && loopItem.value >= maxValue)  && new com.destroystokyo.paper.event.entity.ExperienceOrbMergeEvent((org.bukkit.entity.ExperienceOrb) entity.getBukkitEntity(), (org.bukkit.entity.ExperienceOrb) loopItem.getBukkitEntity()).callEvent()) { // Paper
+-                        if (!loopItem.isRemoved() && !(maxValue > 0 && loopItem.value >= maxValue)) {
++                        if (!loopItem.isRemoved() && !(maxValue > 0 && loopItem.value >= maxValue) && new com.destroystokyo.paper.event.entity.ExperienceOrbMergeEvent((org.bukkit.entity.ExperienceOrb) entity.getBukkitEntity(), (org.bukkit.entity.ExperienceOrb) loopItem.getBukkitEntity()).callEvent()) { // Paper - ExperienceOrbMergeEvent
```
Comment (Prof-Bloodstone) [inline patches/server/0427-ExperienceOrbMergeEvent.patch:20]: I believe the code here duplicates the merging logic that is in net.minecraft.world.entity.ExperienceOrb#merge method - wouldn't it be best to call the event from there?

**Review by Prof-Bloodstone — COMMENTED** [review]
Might be dum questions.

Also, is it worth noting in the API javadocs that the event is not fired if exp orb is about to be spawned but before being added to the world it finds an orb nearby?
See the `award` method calling `tryMergeWithExisting` (or something like that :) )

File: patches/server/0427-ExperienceOrbMergeEvent.patch (hunk 2 above)
Comment (Machine-Maker) [reply patches/server/0427-ExperienceOrbMergeEvent.patch:20]: Yes, you are correct. Probably should put a note of that on the 1.17 issue, as I cant post it there.

**Pull request merged**